plugins/tangoinfo/tangoinfo.py

```diff
@@ -0,0 +1,360 @@
+                    request = WrappedWSGetRequest(self, *args, **kwargs)
+                    return ws.add_request(request)
+                ws.get_patched = get_patched
+                return ws.get_patched(
+                    TANGO_INFO_HOST,
+                    TANGO_INFO_PORT,
+                    path,
+                    partial(self.website_process, barcode, zeros),
+                    priority=False,
+                    important=False,
+                    parse_response_type=None,
+                    queryargs=None,
+                )
+            # Call website_process() as a partial func
+            return album.tagger.webservice.get(
+                TANGO_INFO_HOST,
+                TANGO_INFO_PORT,
+                path,
+                partial(self.website_process, barcode, zeros),
+                priority=False,
+                important=False,
+                parse_response_type=None,
+                queryargs=None,
+            )
+
+    def website_process(self, barcode, zeros, response_bytes, reply, error):
+        """
+        response_bytes: PyQt5.QtCore.QByteArray, equals reply.readAll()
+        reply:          PyQt5.QtNetwork.QNetworkReply
+        error:          PyQt5.QtNetwork.QNetworkReply.NetworkError (optional)
+        """
+
+        if error:
+            log.warning("%s: Network error retrieving info for barcode %s",
+                        PLUGIN_NAME, barcode)
+            track_triple = self.albumpage_queue.pop(barcode)
+            for track, album, tint in track_triple:
+                self.album_remove_request(album)
+            return
+
+        # Decode QByteArray into unicode
+        response_decoded = response_bytes.data().decode('utf-8')
+
+        tangoinfo_albumdata = self.extract_data(barcode, response_decoded)
+
+        self.albumpage_cache[barcode] = tangoinfo_albumdata
+        track_triple = self.albumpage_queue.pop(barcode)
+
+        if tangoinfo_albumdata:
+            if zeros:
+                log.debug("%s: tango.info does not seem to have data for "
+                          "barcode %s. "
+                          "However, retrying with barcode %s (i.e. the same "
+                          "with %s prepended) was successful. "
+                          "This most likely means either MusicBrainz or "
+                          "tango.info has stored a wrong barcode for this "
+                          "release. You might want to investigate this "
+                          "discrepancy and report it.",
+                          PLUGIN_NAME, barcode, ("0" * zeros) + barcode,
+                          ("a zero" if zeros == 1 else "%d zeros" % zeros))
+
+            for track, album, tint in track_triple:
+                tm = track.metadata
+                if not self.albumpage_cache[barcode].get(tint):
+                    self.album_remove_request(album)
+                    continue
+
+                for field in ["genre", "date", "vocal"]:
```

Comment:
Just a minor thing, but this list could be declared as a tuple.
```suggestion
                for field in ("genre", "date", "vocal"):
```

Reply:
Done